crates/biome_js_analyze/src/assist/source/organize_imports.rs

```diff
@@ -1,144 +1,521 @@
-    /// import sibling from "./sibling";      // Import group 1
-    /// import "z";  // Import group 2
-    /// import "a";  // Import group 3
-    /// import assert from "node:assert";  // Import group 4
-    /// import React from "react";         // Import group 4
+    /// // Attached comment for `d`
+    /// import D from "d";
+    /// // Attached comment for `e`
+    /// import E from "e";
+    /// import F from "f";
+    ///
+    /// // Detached comment (new chunk)
+    ///
+    /// // Attached comment for `a`
+    /// import A from "a";
+    /// // Attached comment for `b`
+    /// import B from "b";
     /// ```
+    ///
+    ///
+    /// ## Named imports, named exports and attributes sorting
+    ///
+    /// The sorter also sort named imports, named exports, as well as attributes.
+    /// It uses a natural sort order.
+    ///
+    /// The following code...
+    ///
+    /// ```js,expect_diagnostic
+    /// import { a, b, A, B, c10, c9 } from "a";
+    ///
+    /// export { a, b, A, B, c10, c9 } from "a";
+    ///
+    /// import special from  "special" with { "type": "ty", "metadata": "data" };
+    /// ```
+    ///
+    /// ...is sorted as follows:
+    ///
+    /// ```js,ignore
+    /// import { A, a, B, b, c9, c10 } from "a";
+    ///
+    /// export { A, a, B, b, c9, c10 } from "a";
+    ///
+    /// import special from  "special" with { "metadata": "data", "type": "ty" };
+    /// ```
+    ///
+    ///
+    /// ## Supported glob patterns
+    ///
+    /// You have to understand the structure of a source to understand which source match a glob.
+    /// A source is divided in source segments.
+    /// Every source segment is delimited by the separator `/` or the start/end of the source.
+    /// For instance `src/file.js` consists of two source segments: `src` and `file.js`.
+    ///
+    /// - star `*` that matches zero or more characters inside a source segment
+    ///
+    ///   `file.js` matches `*.js`.
+    ///   Conversely, `src/file.js` doesn't match `*.js`
+    ///
+    /// - globstar `**` that matches zero or more source segments
+    ///   `**` must be enclosed by separators `/` or the start/end of the glob.
+    ///   For example, `**a` is not a valid glob.
+    ///   Also, `**` must not be followed by another globstar.
+    ///   For example, `**/**` is not a valid glob.
+    ///
+    ///   `file.js` and `src/file.js` match `**` and `**/*.js`
+    ///   Conversely, `README.txt` doesn't match `**/*.js` because the source ends with `.txt`.
+    ///
+    /// - Use `\*` to escape `*`
+    ///
+    ///   the source `*` matches `\*`.
+    ///
+    /// - `?`, `[`, `]`, `{`, and `}` must be escaped using `\`.
+    ///   These characters are reserved for possible future use.
+    ///
+    /// - Use `!` as first character to negate a glob
+    ///
+    ///   `file.js` matches `!*.test.js`.
+    ///   `src/file.js` matches `!*.js` because the source contains several segments.
```

Comment:
```suggestion
    ///   `!*.test.js` matches `file.js` because it doesn't end with `.test.js`.
    ///   `!*.js` matches `src/file.js` because the source contains several segments.
```